src/main/java/org/terasology/climateConditions/HyperthermiaSystem.java

```diff
@@ -15,103 +15,47 @@
  */
 package org.terasology.climateConditions;
 
-import org.terasology.biomesAPI.Biome;
-import org.terasology.biomesAPI.BiomeRegistry;
-import org.terasology.biomesAPI.OnBiomeChangedEvent;
 import org.terasology.entitySystem.entity.EntityRef;
-import org.terasology.entitySystem.entity.lifecycleEvents.BeforeRemoveComponent;
-import org.terasology.entitySystem.entity.lifecycleEvents.OnAddedComponent;
 import org.terasology.entitySystem.event.ReceiveEvent;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
 import org.terasology.logic.characters.AffectJumpForceEvent;
-import org.terasology.logic.characters.CharacterMovementComponent;
 import org.terasology.logic.characters.GetMaxSpeedEvent;
 import org.terasology.logic.health.HealthComponent;
 import org.terasology.logic.health.event.ActivateRegenEvent;
 import org.terasology.logic.health.event.ChangeMaxHealthEvent;
-import org.terasology.logic.location.LocationComponent;
-import org.terasology.logic.players.PlayerCharacterComponent;
-import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
-import org.terasology.math.geom.Vector3i;
-import org.terasology.naming.Name;
-import org.terasology.registry.In;
 import org.terasology.thirst.event.AffectThirstEvent;
 
-import java.util.Optional;
-
 @RegisterSystem(value = RegisterMode.AUTHORITY)
 public class HyperthermiaSystem extends BaseComponentSystem {
-    private final Name DesertId = new Name("CoreWorlds:Desert");
-
-    @In
-    BiomeRegistry biomeRegistry;
-
-    @ReceiveEvent(components = {PlayerCharacterComponent.class, CharacterMovementComponent.class})
-    public void onBiomeChange(OnBiomeChangedEvent event, EntityRef player) {
-        if (event.getNewBiome().getId().equals(DesertId)) {
-            player.addOrSaveComponent(new HyperthermiaComponent());
-        } else {
-            if (player.hasComponent(HyperthermiaComponent.class)) {
-                player.removeComponent(HyperthermiaComponent.class);
-            }
-        }
-    }
```

Comment:
How is `HyperthermiaComponent` added without this at the moment?

Reply:
HyperthermiaComponent will be added with the appropriate level in the BodyTemperatureSystem depending on the BodyTemperatureLevel. 
After this PR Hypotherima will no longer be added in a "danger zone". Since, the transition has to take place at some time, I think it would be appropriate to make it here. I think this PR should start working with the BodyTemperature PR..

Reply:
If I understand that correctly in between those two PRs the effects won't work, right? I think it would be a cleaner change to just add and remove the components (with level 1) and do the switch to body temperature system in that PR... 🤔